File: rust/src/models/cancel_checkout_session_response.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::checkout_session::CheckoutSession;

#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct CancelCheckoutSessionResponse {
    pub session: CheckoutSession,
}

impl CancelCheckoutSessionResponse {
    pub fn new(session: CheckoutSession) -> Self {
        Self { session }
    }
}

File: typescript/src/models/cancelCheckoutSessionResponse.ts
// this file is @generated
import { type CheckoutSession, CheckoutSessionSerializer } from "./checkoutSession";

export interface CancelCheckoutSessionResponse {
  session: CheckoutSession;
}

export const CancelCheckoutSessionResponseSerializer = {
  _fromJsonObject(object: any): CancelCheckoutSessionResponse {
    return {
      session: CheckoutSessionSerializer._fromJsonObject(object["session"]),
    };
  },

  _toJsonObject(self: CancelCheckoutSessionResponse): any {
    return {
      session: CheckoutSessionSerializer._toJsonObject(self.session),
    };
  },
};
